src/Xenial.Tasty/Commands/ExecuteTestsCommand.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Xenial.Delicious.Execution;
+using Xenial.Delicious.Metadata;
+using Xenial.Delicious.Scopes;
+using Xenial.Delicious.Visitors;
+
+namespace Xenial.Delicious.Commands
+{
+    public static class ExitCommand
+    {
+        public static (string name, Func<RuntimeContext, Task> command, string? description, bool? isDefault) Register()
+            => ("x", Execute, "Exit interactive run", false);
+
+        static Task Execute(RuntimeContext context)
+        {
+            context.EndPipeLine = true;
+            return Task.CompletedTask;
+        }
+    }
+
+    public static class ExecuteTestsCommand
+    {
+        public static (string name, Func<RuntimeContext, Task> command, string? description, bool? isDefault) Register()
+            => ("e", Execute, "Execute all tests in default order", true);
+
+        static async Task Execute(RuntimeContext context)
+        {
+            context.TestQueue = await Execute(context.Executor);
+            context.TestQueue = await VisitForcedTestCases(context.TestQueue);
+            await Execute(context.Executor, context.TestQueue);
+        }
+
+        static async Task<Queue<TestCase>> Execute(TestExecutor executor)
+        {
+            var groupQueue = new Queue<TestGroup>(executor.Scope.Descendants().OfType<TestGroup>());
+            var testQueue = new Queue<TestCase>();
+
+            while (groupQueue.Count > 0)
+            {
+                var currentGroup = groupQueue.Dequeue();
+                await Execute(executor, groupQueue, testQueue, currentGroup);
+            }
+
+            return testQueue;
+        }
+
+        static async Task Execute(TestExecutor executor, Queue<TestGroup> groupQueue, Queue<TestCase> testQueue, TestGroup testGroup)
+        {
+            var app = executor.BuildTestGroupMiddleware();
+            var context = new TestGroupContext(testGroup, executor.Scope, groupQueue, testQueue);
+            await app(context);
+        }
+
+        public static Task<Queue<TestCase>> VisitForcedTestCases(Queue<TestCase> testQueue)
```

Comment:
really should not be public. But for custom commands this could be really valuable to users though. We need a more high level API for this I guess.